lib/govuk_markdown/preprocessor.rb

```diff
@@ -7,7 +7,7 @@ def initialize(document)
     end
 
     def inject_inset_text
-      output.gsub!(build_regexp("inset-text")) do
+      @output = output.gsub(build_regexp("inset-text")) do
```

Comment:
is there a reason why you're using the @ here?

Reply:
Yeah - we want to [overwrite the instance variable](https://ruby-doc.org/docs/ruby-doc-bundle/UsersGuide/rg/instancevars.html) here without updating it in place, which results in an error when the string is frozen.

Reply:
mmmmmm i see! So even though it's defined in the attr_reader we need to use @ ?

Reply:
have seen your slack reply! got it now